paginator/paginator_paginate_test.go

```diff
@@ -669,6 +674,134 @@ func (s *paginatorSuite) TestPaginateCustomTypeString() {
 	s.assertIDs(p1, 9, 8, 7)
 }
 
+type NullString sql.NullString
```

Comment:
I don't quite get the concept of using the null string, to me `''`, `'null'` are all valid string 🙈

What about making a nullable field in `JSON` type, and using it as pagination key? For example, setting one `order` with `Data.keyInt` equal to `nil`, and other orders with valid numbers. It would be clearer to see the ordering effect.

Reply:
Fair point, I'm sure there's other options of implementing this, but `NullString` can be useful (compared to `*string`) in cases where you want to differentiate between a `nil` Go variable and a `NULL` database value. I guess it's actually used relatively widely. In our case we even extend it with another boolean field so that we can differentiate between values that have been set (by an API update handler, or by retrieving from the DB) and unset values.

Anyhow, I don't think my example is valid with a `JSON` type. If you use `Data.keyInt` as the pagination key and set `CustomType.Type` to a "primitive" (int), the decoder will assign it the default value of the type (`0`) when decoding from `null` in the cursor. If `CustomType.Type` is (more correctly) set to a pointer, the pointer will be `nil` and it will be correctly recognized by the existing `isNil` function, so the problem I'm describing does not manifest.
It really only applies to cases where custom type is a struct.

I suppose we could represent a struct with a JSON object to reuse the existing `Data` field in `order`, but it seems like we still wouldn't be able to avoid a definition of a new type and the bunch of methods that come with it.

Reply:
Thanks for the explanation 🙌 I think I could get your point, and I might need some time to think more about it. 🧐 Will get back to you soon.

Reply:
After fully understanding and experiment, I agree with you that null string maybe the best approach to this case. Just leaving some minor suggestions you can take a look when free 😀